lib/json/jwe.rb

```diff
@@ -40,8 +40,8 @@ def encrypt!(public_key_or_secret)
     end
 
     def decrypt!(private_key_or_secret, algorithms = nil, encryption_methods = nil)
-      raise UnexpectedAlgorithm.new('Unexpected alg header') unless algorithms.blank? || Array(algorithms).include?(alg)
-      raise UnexpectedAlgorithm.new('Unexpected enc header') unless encryption_methods.blank? || Array(encryption_methods).include?(enc)
+      raise UnexpectedAlgorithm.new('Unexpected alg header') unless algorithms.nil? || Array(algorithms).include?(alg)
```

Comment:
This is not equivalent to original code.

On the other hand, `Array(algorithms).include?(alg)` should be sufficient, as `Array(nil)` returns an empty array, so `nil` or an "empty" object would have to be passed in an array explicitly to pass the check. At which point it's best to let the foot do the shooting.